src/ledger/LedgerManagerImpl.cpp

```diff
@@ -739,17 +739,35 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     // step 4
     mApp.getBucketManager().forgetUnreferencedBuckets();
 
-    // Maybe sleep for parameterized amount of time in simulation mode
-    auto sleepFor = std::chrono::microseconds{
-        mApp.getConfig().OP_APPLY_SLEEP_TIME_FOR_TESTING * txSet->sizeOp()};
-    std::chrono::microseconds applicationTime =
-        closeLedgerTime.checkElapsedTime();
-    if (applicationTime < sleepFor)
+    if (!mApp.getConfig().OP_APPLY_SLEEP_TIME_FOR_TESTING.empty())
     {
-        sleepFor -= applicationTime;
-        CLOG_DEBUG(Perf, "Simulate application: sleep for {} microseconds",
-                   sleepFor.count());
-        std::this_thread::sleep_for(sleepFor);
+        // Sleep for a parameterized amount of time in simulation mode
+        int sample = rand_uniform(1, 100);
+        uint32_t chosenDuration = 0;
```

Comment:
extract in a separate function, also, why not just have `OP_APPLY_SLEEP_TIME_FOR_TESTING` be exactly 100 elements, that way there is no need for a loop here and just do a random pick

Reply:
Put the logic in `chooseOpApplySleepTimeForTesting` and turned `OP_APPLY_SLEEP_TIME_FOR_TESTING` into a vector of 100 elements